From e19ac63f03c4e5103f1456803b5e1db8169b9040 Mon Sep 17 00:00:00 2001
From: Matthew Feickert <matthew.feickert@cern.ch>
Date: Sun, 9 Feb 2025 03:10:29 -0700
Subject: [PATCH] fix: Make makefile_ML5_lib compatible with precompiled
 libraries

* Let makefile_ML5_lib build IREGI standalone against existing
  compiled versions of the libraries.
---
 vendor/IREGI/src/makefile_ML5_lib | 17 +++--------------
 1 file changed, 3 insertions(+), 14 deletions(-)

diff --git a/vendor/IREGI/src/makefile_ML5_lib b/vendor/IREGI/src/makefile_ML5_lib
index fff9dc8..7319db6 100644
--- a/vendor/IREGI/src/makefile_ML5_lib
+++ b/vendor/IREGI/src/makefile_ML5_lib
@@ -6,14 +6,9 @@ FFDIR=$(qcdlooppath)/ff
 onelooppath=$(PWD)/oneloop
 #FFLAGS=-O -g -ggdb -fno-automatic -fbounds-check -I$(onelooppath) # -fbounds-check is very useful in checking mermory error,-g is for debuging,-ggdb is for debuging special in gdb
 FFLAGS=-O -g -fPIC -I$(onelooppath)
-obj= global.o funlib.o linear_algebra.o kinematics.o matrices.o matrix_base.o cmatrix_base.o mis_warp.o special_fun.o pave_reduce.o cpave_reduce.o si_reduce.o csi_reduce.o ti_reduce.o cti_reduce.o gti_reduce.o binary_tree.o IREGI4ML5_interface.o avh_olo_foriregi.o
-olomod= avh_olo_foriregi.mod avh_olo_foriregi_dp_box.mod    avh_olo_foriregi_dp_olog.mod   avh_olo_foriregi_units.mod \
-        avh_olo_foriregi_dp.mod        avh_olo_foriregi_dp_boxc.mod   avh_olo_foriregi_dp_prec.mod   avh_olo_foriregi_version.mod \
-        avh_olo_foriregi_dp_arrays.mod avh_olo_foriregi_dp_bub.mod    avh_olo_foriregi_dp_print.mod \
-        avh_olo_foriregi_dp_auxfun.mod avh_olo_foriregi_dp_dilog.mod  avh_olo_foriregi_dp_qmplx.mod \
-        avh_olo_foriregi_dp_bnlog.mod  avh_olo_foriregi_dp_kinds.mod  avh_olo_foriregi_dp_tri.mod
-mod= global.mod funlib.mod linear_algebra.mod kinematics.mod matrices.mod matrix_base.mod cmatrix_base.mod mis_warp.mod special_fun.mod pave_reduce.mod cpave_reduce.mod si_reduce.mod csi_reduce.mod ti_reduce.mod cti_reduce.mod gti_reduce.mod $(olomod)
-libiregi.a: make_ql $(obj) $(mod)
+obj= global.o funlib.o linear_algebra.o kinematics.o matrices.o matrix_base.o cmatrix_base.o mis_warp.o special_fun.o pave_reduce.o cpave_reduce.o si_reduce.o csi_reduce.o ti_reduce.o cti_reduce.o gti_reduce.o binary_tree.o IREGI4ML5_interface.o
+mod= global.mod funlib.mod linear_algebra.mod kinematics.mod matrices.mod matrix_base.mod cmatrix_base.mod mis_warp.mod special_fun.mod pave_reduce.mod cpave_reduce.mod si_reduce.mod csi_reduce.mod ti_reduce.mod cti_reduce.mod gti_reduce.mod
+libiregi.a: $(obj) $(mod)
 	@rm -f libiregi.a
 	@ar csr libiregi.a  $(obj)
 #$(QLDIR)libqcdloop.a $(FFDIR)libff.a $(onelooppath)libavh_olo.a
@@ -39,8 +34,6 @@ matrix_base.o matrix_base.mod : matrix_base.f90
 	$(FC) -c $< $(FFLAGS)
 cmatrix_base.o cmatrix_base.mod : cmatrix_base.f90
 	$(FC) -c $< $(FFLAGS)
-avh_olo_foriregi.mod : oneloop/avh_olo_foriregi.f90
-	$(FC) -c $< $(FFLAGS)
 mis_warp.o mis_warp.mod : mis_warp.f90 kinematics.mod global.mod $(olomod)
 	$(FC) -c $< $(FFLAGS)
 special_fun.o special_fun.mod : special_fun.f90
@@ -64,9 +57,5 @@ binary_tree.o binary_tree.mod : binary_tree.f90 global.mod
 IREGI4ML5_interface.o : IREGI4ML5_interface.f90 global.mod funlib.mod ti_reduce.mod binary_tree.mod cti_reduce.mod
 	$(FC) -c $< $(FFLAGS)
 
-make_ql:
-	(make -f makefile_qcdloop)
-
 clean:
 	@rm -rf $(obj) $(exe) *.mod *~ *.a
-	@(make clean -f makefile_qcdloop)
-- 
2.47.1

